### PR TITLE
add support for extract(field from source)

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: docs/sql/scalar.txt [hunk 1]
```diff
@@ -144,11 +144,82 @@ If the time zone is not specified, truncation is based on UTC time::
     +---------------+---------------+
     SELECT 3 rows in set (... sec)
 
+extract(field from source) returns long
```
Comment (mfelsche) [inline docs/sql/scalar.txt]: extract is no scalar function but an expression like match, maybe list it elsewhere in a separate section?

**Comment by mfelsche** [comment]
change documentation and wording for EXTRACT.
otherwise ack.

File: docs/sql/scalar.txt (hunk 1 above)
Comment (mfussenegger) [inline docs/sql/scalar.txt]: Would leave it here, most other docs refer to it as function too.

See 

- http://www.postgresql.org/docs/9.1/static/functions-datetime.html
- http://docs.oracle.com/cd/B19306_01/server.102/b14200/functions050.htm
- http://www.w3schools.com/sql/func_extract.asp

File: docs/sql/scalar.txt [hunk 2]
```diff
@@ -144,11 +144,113 @@ If the time zone is not specified, truncation is based on UTC time::
     +---------------+---------------+
     SELECT 3 rows in set (... sec)
 
+extract(field from source)
+--------------------------
+
+``extract`` is a special expression that translates to a function which
+retrieves subfields such as day, hour or minute from a timestamp.
+
+The return type is dependend on the used ``field``.
```
Comment (mfelsche) [inline docs/sql/scalar.txt]: typo: depends on

**Pull request merged**